src/sqlacodegen/generators.py

```diff
@@ -1201,18 +1202,28 @@ def render_column_attribute(self, column_attr: ColumnAttribute) -> str:
         column = column_attr.column
         rendered_column = self.render_column(column, column_attr.name != column.name)
 
-        try:
-            python_type = column.type.python_type
+        def recursively_get_col_type(column_type: TypeEngine[Any]) -> str:
```

Comment:
Thinking about it a little further doing a whole recursive clojure is a bit...too much.

It sure looks neat, but I can make it a more boring `if` condition if the maintainers think it's a good idea.

Reply:
Let's start with a boring `if` variation and move from there. SQL (PG) arrays is a very specific beast.  I don't see us currently supporting complex collections with nested generic types (not really a thing in RDBMSs)

Reply:
Great :clap: